atlp-rwanda/eagles-ec-be
# #187419151-Initial Project Setup with TypeScript, PostgreSQL, and Sequelize

#### What does this PR do?

This PR sets up the initial project structure with TypeScript, PostgreSQL for the database, and Sequelize as the ORM. It includes the creation of a dummy user model in the database and sets up basic server functionality.

#### Description of Task to be completed?

- Initialize project with TypeScript, PostgreSQL, and Sequelize.
- Create folder structure for the project.
- Implement a basic user model in the database.
- Set up a basic home route ("/") to confirm server functionality.

#### How should this be manually tested?

1. Run `npm run migrate` to execute the initial database migration and create the test tables in db.
2. Run `npm run seed` to seed the database with dummy user data.
3. Start the development server using `npm run dev`.
4. Access the home route ("/") in your browser or API testing tool to verify server response.
5. Check the user route ("/api/v1/users") to confirm that dummy user data is accessible.

#### Any background context you want to provide?

This PR is part of the initial project setup. It includes the configuration of essential technologies and the creation of foundational components for further development.

#### What are the relevant pivotal tracker/Trello stories?
#187419151
#### Screenshots 
<img width="314" alt="Screenshot 2024-04-13 at 13 05 37" src="https://github.com/atlp-rwanda/eagles-ec-be/assets/105858896/9aa9b9e5-c08e-472a-918f-423f6f390f4f">
<img width="440" alt="Screenshot 2024-04-13 at 13 16 46" src="https://github.com/atlp-rwanda/eagles-ec-be/assets/105858896/bae0cc61-555d-4a9c-9c2b-1850c4904f76">
